downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,211 @@
+import org.apache.logging.log4j.Logger;
+import org.biopax.validator.api.Validator;
+import org.biopax.validator.api.ValidatorUtils;
+import org.biopax.validator.api.beans.Validation;
+import org.biopax.validator.impl.IdentifierImpl;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.reactome.biopax.SpeciesAllPathwaysConverter;
+import org.reactome.biopax.SpeciesAllPathwaysLevel3Converter;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+
+public class Biopax {
+  private static final Logger logger = LogManager.getLogger();
+  static ApplicationContext ctx;
+  static boolean autofix = false;
+  static int maxErrors = 0;
+  static String profile = "notstrict";
+  static String outFormat = "xml";
+  static Validator validator;
+
+  // BioPAX validation requires loading of bio ontologies, which takes a few minutes before starting the BioPAX process
+  static {
+   logger.info("Running BioPAX step");
+   logger.info("Preparing BioPAX validation rules...");
+    ctx = new ClassPathXmlApplicationContext(new String[]{
+      "META-INF/spring/appContext-validator.xml",
+      "META-INF/spring/appContext-loadTimeWeaving.xml"
+    });
+    validator = (Validator) ctx.getBean("validator");
+    logger.info("Finished preparing BioPAX validation rules");
+  }
+  
+  // Once BioPAX validation rules have loaded, the actual BioPAX process can start
+  public static void execute(String username, String password, String host, String port, String database, String releaseNumber, String pathToSpeciesConfig, boolean runBioPAX2, boolean runBioPAX3) throws Exception {
+	  	// Temporary location where BioPAX files are stored
+	  	String biopaxDir = releaseNumber + "_biopax";
+        File biopaxDirFile = new File(biopaxDir);
+        if (!biopaxDirFile.exists()) {
+        	biopaxDirFile.mkdir();
+        }
+        
+        JSONParser parser = new JSONParser();
+        JSONObject speciesFile = (JSONObject) parser.parse(new FileReader(pathToSpeciesConfig));
+        @SuppressWarnings("unchecked")
+		Set<String> speciesKeys = speciesFile.keySet();
+        // First, generate owl files. This particular step requires a local maven installation of the PathwayExchange jar (see README.md for DownloadDirectory)
+        // We want to run BioPAX level 2 and BioPAX level 3, so the below loop starts with '2'
+		for (int biopaxLevel = 2; biopaxLevel < 4; biopaxLevel++) {
```

Comment:
It might be better to extract the body of the for loop into a method and call it twice with each biopaxLevel.  Having a loop starting at 2 that iterates twice is a little confusing.

You could also do:

```
List<Integer> biopaxLevels = Arrays.asList(2,3);
for (int biopaxLevel: biopaxLevels) {
...
}
```

Reply:
Yes thats the clean solution I'd been waiting for -- I'll implement that